hathor/manager.py

```diff
@@ -1013,16 +1029,18 @@ def on_new_tx(self, tx: BaseTransaction, *, conn: Optional[HathorProtocol] = Non
             tx.update_initial_metadata()
             self.tx_storage.save_transaction(tx)
 
-        if tx.is_transaction:
+        if tx.is_transaction and self._enable_sync_v2:
```

Comment:
Should we check for `self.tx_storage.indexes.deps is not None` instead of sync v2?

Reply:
I avoided using that as a condition, at least in most places withing the HathorManager, because I want to assert for the index.

Reply:
Thanks!